### ISSUE TITLE
List Dialog Controller setting

### ISSUE THREAD

**Issue by Swiftb0y** [mixxxdj/mixxx#14084]
### Feature Description

It would be very nice to have a dialog that lets the user construct a list out of a given set of options. That dialog would let the user choose between a number of options and rearrange them in their preferred order. I'm sure there is a name for this type of widget, I just don't know it. I would imagine it to be similar to the `Preferences > Effects > Visible Effects` widget. 
![image](https://github.com/user-attachments/assets/91d489f7-c7b2-4b96-bd33-8de53382c98a)

CC @acolombier wdyt?

**Comment by Swiftb0y** [comment]
The primary usecase for this is to model cases where a combobox would not make sense due to combinatorial explosion. For example letting the user create a bunch of `rate_ranges` or their preferred setup for the `parameter` (EQ) buttons.

**Comment by acolombier** [comment]
I think that makes sense. As such, this could be an extension of the existing settings types, so each type could allow a multiple selection, similar to the `multiple` attributes of the [`select` HTML input field](https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_select_multiple). I guess we could also add an `orderable` attribute, which could make the `rate_ranges` usecase more user friendly as well, does that make sense?

As such, the widget design could look similar to GH label selection (in this case, it is quite similar to `enum` behaviour, but the input field could obviously be a number, colour or file as well)
![image](https://github.com/user-attachments/assets/5d4ddf86-7bf0-45ef-b5f8-514f0f37eaa9)

Cannot remember the name of this type of widget UX either tho

**Comment by Swiftb0y** [comment]
yeah that could also work, though I'm not sure about good the ordering UX is on that type of widget. In either case, it would be cool if it could be generic over the actual enclosed widget. So you could stuff whatever widget you'd like in there.

**Comment by Swiftb0y** [comment]
In addition to an `orderable` (which is not a word, so I'd prefer something like `ordered`) attribute, a `element_count_min` and `element_count_max` would be nice too.
